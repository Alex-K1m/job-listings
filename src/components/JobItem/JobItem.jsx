import cn from 'classnames';
import { nanoid } from 'nanoid';
import { arrayOf, bool, func, shape, string } from 'prop-types';
import React from 'react';

import Chip from '../Chip';
import Logo from '../Logo';
import Tag from '../Tag';
import styles from './JobItem.module.scss';

const JobItem = ({ data, onTagClick }) => {
  const {
    company,
    logoUrl,
    new: isNew,
    featured: isFeatured,
    position,
    postedAt,
    contract,
    location,
    tags,
  } = data;

  return (
    <div className={cn(styles.job, { [styles.jobHighlighted]: isFeatured })}>
      <div className={styles.logo}>
        <Logo src={logoUrl} alt='Photosnap logo' />
      </div>
      <div className={styles.body}>
        <div className={styles.heading}>
          <h2>{company}</h2>
          {isNew && <Chip>New!</Chip>}
          {isFeatured && <Chip color='dark'>Featured</Chip>}
        </div>
        {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
        <a href='#' className={styles.position}>
          <h3>{position}</h3>
        </a>
        <div className={styles.details}>
          <span>{postedAt}</span>
          <span>{contract}</span>
          <span>{location}</span>
        </div>
      </div>
      <hr className={styles.hr} />
      <ul className={styles.tagList}>
        {tags.map((label) => (
          <li className={styles.tagListItem} key={nanoid()}>
            <Tag onClick={onTagClick(label)}>{label}</Tag>
          </li>
        ))}
      </ul>
    </div>
  );
};

JobItem.propTypes = {
  data: shape({
    company: string.isRequired,
    logoUrl: string.isRequired,
    new: bool,
    featured: bool,
    position: string.isRequired,
    postedAt: string.isRequired,
    contract: string.isRequired,
    location: string.isRequired,
    tags: arrayOf(string),
  }).isRequired,
  onTagClick: func.isRequired,
};

export default JobItem;
