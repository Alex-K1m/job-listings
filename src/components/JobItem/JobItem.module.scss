@use '../colors' as c;
@use '../breakpoints' as brp;

$logo-size-xs: 4.8rem;
$padding-xs: 2.4rem;

.job {
  background-color: #fff;
  border-radius: 0.5rem;
  box-shadow: 0 1.5rem 2rem -0.5rem c.$shadow;
  margin-top: $logo-size-xs / 2;
  padding: $padding-xs;

  @include brp.sm-up {
    align-items: center;
    display: flex;
    margin-top: 0;
    padding: 3.2rem 4rem;
  }

  &Highlighted {
    background-image: linear-gradient(to right,
    c.$primary 0.5rem,
    transparent 0.5rem,);
  }
}

.logo {
  height: $logo-size-xs;
  margin-top: -$logo-size-xs / 2 - $padding-xs;
  width: $logo-size-xs;

  @include brp.sm-up {
    flex-shrink: 0;
    height: 8.8rem;
    margin-top: 0;
    width: 8.8rem;
  }
}

.body {
  margin-top: 0.8rem;

  @include brp.sm-up {
    flex: 0 1 31rem;
    margin-left: 2.4rem;
    margin-right: 1.6rem;
    margin-top: 0;
  }
}

.heading {
  align-items: center;
  color: c.$primary;
  display: flex;
  flex-flow: wrap;
  font: 700 1.3rem/190% Spartan, sans-serif;
  margin: -0.2rem -0.4rem;

  @include brp.sm-up {
    font-size: 1.4rem;
    line-height: 175%;
  }

  > :first-child {
    margin: 0.2rem 1.2rem 0.2rem 0.4rem;
  }

  > :not(:first-child) {
    margin: 0.2rem 0.4rem;
  }
}

.position {
  color: c.$text;
  display: block;
  font: 700 1.5rem/160% Spartan, sans-serif;
  margin-top: 0.8rem;
  text-decoration: none;

  @include brp.sm-up {
    font-size: 1.8rem;
    line-height: 135%;
  }

  &:hover {
    color: c.$primary;
  }
}

.details {
  color: c.$text-secondary;
  font: 500 1.3rem/185% Spartan, sans-serif;
  letter-spacing: -0.0077em;
  margin-top: 0.8rem;

  @include brp.sm-up {
    font-size: 1.5rem;
    line-height: 160%;
  }

  > :not(:first-child)::before {
    background-color: c.$text-secondary;
    border-radius: 50%;
    content: '';
    display: inline-block;
    height: 0.4rem;
    margin: 0 0.8rem;
    vertical-align: middle;
    width: 0.4rem;

    @include brp.sm-up {
      margin: 0 1.6rem;
    }
  }
}

.hr {
  background-color: c.$text-secondary;
  border: 0;
  height: 0.1rem;
  margin: 1.6rem 0;
  width: 100%;

  @include brp.sm-up {
    display: none;
  }
}

.tagList {
  display: flex;
  flex-flow: wrap;
  margin: -0.8rem;

  @include brp.sm-up {
    flex: 1 1 26rem;
    justify-content: flex-end;
    margin-left: auto;
  }
}

.tagListItem {
  margin: 0.8rem;
}
