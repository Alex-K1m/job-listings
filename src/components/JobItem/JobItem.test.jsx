import React from 'react';
import renderer from 'react-test-renderer';

import jobs from '../../data.json';
import JobItem from './JobItem';

describe('JobItem', () => {
  it('renders correctly', () => {
    const job = jobs[0];
    const { role, level, languages, tools } = job;

    job.logoUrl = 'https://example.com';
    job.tags = [role, level, ...languages, ...tools];

    const json = renderer
      .create(<JobItem data={jobs[0]} onTagClick={jest.fn()} />)
      .toJSON();

    expect(json).toMatchSnapshot();
  });
});
