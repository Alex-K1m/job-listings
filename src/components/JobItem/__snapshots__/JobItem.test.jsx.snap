// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobItem renders correctly 1`] = `
<div
  className="job jobHighlighted"
>
  <div
    className="logo"
  >
    <img
      alt="Photosnap logo"
      className="logo"
      src="https://example.com"
    />
  </div>
  <div
    className="body"
  >
    <div
      className="heading"
    >
      <h2>
        Photosnap
      </h2>
      <div
        className="chip"
      >
        <span
          className="label"
        >
          New!
        </span>
      </div>
      <div
        className="chip dark"
      >
        <span
          className="label"
        >
          Featured
        </span>
      </div>
    </div>
    <a
      className="position"
      href="#"
    >
      <h3>
        Senior Frontend Developer
      </h3>
    </a>
    <div
      className="details"
    >
      <span>
        1d ago
      </span>
      <span>
        Full Time
      </span>
      <span>
        USA only
      </span>
    </div>
  </div>
  <hr
    className="hr"
  />
  <ul
    className="tagList"
  >
    <li
      className="tagListItem"
    >
      <div
        className="wrapper"
      >
        <div
          className="tag"
        >
          Frontend
        </div>
      </div>
    </li>
    <li
      className="tagListItem"
    >
      <div
        className="wrapper"
      >
        <div
          className="tag"
        >
          Senior
        </div>
      </div>
    </li>
    <li
      className="tagListItem"
    >
      <div
        className="wrapper"
      >
        <div
          className="tag"
        >
          HTML
        </div>
      </div>
    </li>
    <li
      className="tagListItem"
    >
      <div
        className="wrapper"
      >
        <div
          className="tag"
        >
          CSS
        </div>
      </div>
    </li>
    <li
      className="tagListItem"
    >
      <div
        className="wrapper"
      >
        <div
          className="tag"
        >
          JavaScript
        </div>
      </div>
    </li>
  </ul>
</div>
`;
